stdlib/public/core/LifetimeManager.swift

```diff
@@ -112,6 +112,35 @@ public func withUnsafeMutablePointer<T, Result>(
 ///
 /// The pointer argument to `body` is valid only during the execution of
 /// `withUnsafePointer(to:_:)`. Do not store or return the pointer for later
+/// use. It is undefined behavior to attempt to mutate through the
+///
+/// - Parameters:
+///   - arg: An instance to temporarily use via pointer.
+///   - body: A closure that takes a pointer to `arg` as its sole argument. If
+///     the closure has a return value, that value is also used as the return
+///     value of the `withUnsafePointer(to:_:)` function. The pointer argument
+///     is valid only for the duration of the function's execution.
+///     It is undefined behavior to try to mutate through the pointer argument
+///     by converting it to `UnsafeMutablePointer` or any other mutable pointer
+///     type. If you need to mutate the argument through the pointer, use
+///     `withUnsafeMutablePointer` instead.
+/// - Returns: The return value, if any, of the `body` closure.
+@_inlineable
+public func withUnsafePointer<T, Result>(
+  to arg: T,
+  _ body: (UnsafePointer<T>) throws -> Result
+) rethrows -> Result
+{
+  return try body(UnsafePointer<T>(Builtin.addressOfBorrow(arg)))
+}
+
+/// Invokes the given closure with a pointer to the given argument.
+///
+/// The `withUnsafePointer(to:_:)` function is useful for calling Objective-C
+/// APIs that take in parameters by const pointer.
+///
+/// The pointer argument to `body` is valid only during the execution of
+/// `withUnsafePointer(to:_:)`. Do not store or return the pointer for later
```

Comment:
Can you add the parameters to the function name where this is mentioned? `withUnsafeMutablePointer(to:_:)`